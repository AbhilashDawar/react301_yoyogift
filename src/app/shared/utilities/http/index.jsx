import axios from 'axios';

const GET = (endpoint) => {
    return axios.get(endpoint);
}

const POST = (endpoint, payload) => {
    return axios.post(endpoint, payload);
}

const PUT = (endpoint, payload) => {
    return axios.put(endpoint, payload);
}

const PATCH = (endpoint, payload) => {
    return axios.patch(endpoint, payload);
}

const DELETE = (endpoint) => {
    return axios.delete(endpoint);
}

export default {
    GET,
    POST,
    PUT,
    PATCH,
    DELETE
}
