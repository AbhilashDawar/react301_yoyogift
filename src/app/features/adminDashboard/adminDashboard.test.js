import React from 'react';
import { shallow, mount } from 'enzyme';
import AdminDashboard from './index.jsx';
import renderer from 'react-test-renderer';
import InfoBox from '../../shared/components/infoBox';
import http from '../../shared/utilities/http';

describe('<AdminDashboard />', () => {
    it('should have one Button element', () => {
        const component = shallow((<AdminDashboard />));
        expect(component.find(InfoBox)).toHaveLength(4);
    });

    it('should mount and check useEffect hook', () => {
        jest.mock('http');
        const component = mount((<AdminDashboard />));
        expect(component.find(InfoBox)).toHaveLength(4);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<AdminDashboard />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
