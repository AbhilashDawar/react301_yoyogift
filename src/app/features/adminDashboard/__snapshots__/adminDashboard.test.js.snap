// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminDashboard /> renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-adminContainer-1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-216 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root makeStyles-container-217 MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-iconSpan-222 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-icon-223"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 6h-2.18c.11-.31.18-.65.18-1 0-1.66-1.34-3-3-3-1.05 0-1.96.54-2.5 1.35l-.5.67-.5-.68C10.96 2.54 10.05 2 9 2 7.34 2 6 3.34 6 5c0 .35.07.69.18 1H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-5-2c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zM9 4c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm11 15H4v-2h16v2zm0-5H4V8h5.08L7 10.83 8.62 12 11 8.76l1-1.36 1 1.36L15.38 12 17 10.83 14.92 8H20v6z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root makeStyles-content-219 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                className="MuiGrid-root makeStyles-fullContainer-218 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h6
                    className="MuiTypography-root makeStyles-text-221 MuiTypography-h6"
                  >
                    Gift Coupons Available
                  </h6>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h5
                    className="MuiTypography-root makeStyles-number-220 MuiTypography-h5"
                  >
                    0
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-224 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root makeStyles-container-225 MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-iconSpan-230 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-icon-231"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 6h-2.18c.11-.31.18-.65.18-1 0-1.66-1.34-3-3-3-1.05 0-1.96.54-2.5 1.35l-.5.67-.5-.68C10.96 2.54 10.05 2 9 2 7.34 2 6 3.34 6 5c0 .35.07.69.18 1H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-5-2c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zM9 4c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm11 15H4v-2h16v2zm0-5H4V8h5.08L7 10.83 8.62 12 11 8.76l1-1.36 1 1.36L15.38 12 17 10.83 14.92 8H20v6z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root makeStyles-content-227 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                className="MuiGrid-root makeStyles-fullContainer-226 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h6
                    className="MuiTypography-root makeStyles-text-229 MuiTypography-h6"
                  >
                    Gift Coupons Sent
                  </h6>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h5
                    className="MuiTypography-root makeStyles-number-228 MuiTypography-h5"
                  >
                    0
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-232 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root makeStyles-container-233 MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-iconSpan-238 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-icon-239"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root makeStyles-content-235 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                className="MuiGrid-root makeStyles-fullContainer-234 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h6
                    className="MuiTypography-root makeStyles-text-237 MuiTypography-h6"
                  >
                    Registered Users
                  </h6>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h5
                    className="MuiTypography-root makeStyles-number-236 MuiTypography-h5"
                  >
                    0
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-240 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root makeStyles-container-241 MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-iconSpan-246 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-icon-247"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0zm15.35 6.41l-1.77-1.77c-.2-.2-.51-.2-.71 0L6 11.53V14h2.47l6.88-6.88c.2-.19.2-.51 0-.71z"
                  fill="none"
                />
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 14v-2.47l6.88-6.88c.2-.2.51-.2.71 0l1.77 1.77c.2.2.2.51 0 .71L8.47 14H6zm12 0h-7.5l2-2H18v2z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root makeStyles-content-243 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                className="MuiGrid-root makeStyles-fullContainer-242 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h6
                    className="MuiTypography-root makeStyles-text-245 MuiTypography-h6"
                  >
                    Total Reviews Given
                  </h6>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h5
                    className="MuiTypography-root makeStyles-number-244 MuiTypography-h5"
                  >
                    0
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="makeStyles-footer-203"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        className="MuiGrid-root makeStyles-footerMain-204 MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiGrid-root makeStyles-footerDetails-206 MuiGrid-container"
          >
            <span>
              Yo Yo Gift
            </span>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-footerDetails-206 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <span>
                YoYoGift Technologies Pvt. Ltd.
              </span>
              <br />
              <span>
                Global Village Tech Park,
              </span>
              <br />
              <span>
                RR Nagar, Bengaluru 560059
              </span>
              <br />
              <span>
                Karnakata, INDIA
              </span>
            </div>
            <div
              className="MuiGrid-root makeStyles-footerDetails-206 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <span>
                +91-42444 42444
              </span>
              <br />
              <br />
              <span>
                info@yoyogift.in
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-footerReservedText-207 MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <span>
          ©2019 All Rights Reserved
        </span>
      </div>
    </div>
  </footer>,
]
`;
